### PR TITLE
Remove telemetry for unhandled errors

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

The telemetry for unhandled errors are a LOT, so removing them for now. I didn't uncomment it out as I don't quite see another way we could enable it in the future.

## Testing

It's a one-line removal, so should still work like before.

## Docs

n/a. not docs related.

### CODE REVIEW

File: packages/astro/src/vite-plugin-astro-server/error.ts [hunk 1]
```diff
@@ -24,8 +23,6 @@ export function recordServerError(
 	// Our error should already be complete, but let's try to add a bit more through some guesswork
 	const errorWithMetadata = collectErrorMetadata(err, config.root);
 
-	telemetry.record(eventError({ cmd: 'dev', err: errorWithMetadata, isFatal: false }));
```
Comment (matthewp) [inline packages/astro/src/vite-plugin-astro-server/error.ts:27]: this is going to remove telemetry for all errors that happen during a request. Do we just want to remove the unhandled rejections? if so we can check that `_err?.name === 'UnhandledRejection'`

Comment (bluwy) [reply packages/astro/src/vite-plugin-astro-server/error.ts:27]: I didn't know we assigned a specific name to it. Yeah I think that works too 👍

**Review by matthewp — APPROVED** [review]

File: packages/astro/src/vite-plugin-astro-server/error.ts [hunk 2]
```diff
@@ -24,7 +24,10 @@ export function recordServerError(
...
+	// Ignore unhandled rejection errors as they appear A LOT and we cannot record the amount to telemetry
+	if (errorWithMetadata.name !== AstroErrorData.UnhandledRejection.name) {
```
Comment (Princesseuh) [inline packages/astro/src/vite-plugin-astro-server/error.ts:28]: I don't remember, do we need to be defensive against the error not having a name here?

**Review by Princesseuh — APPROVED** [review]
Just one comment, jsut wondering about it., but looks fine to me otherwise. Would love to figure out what issues people were hitting though, but we don't have much info to go on from, unfortunately

File: packages/astro/src/vite-plugin-astro-server/error.ts (hunk 2 above)
Comment (matthewp) [reply packages/astro/src/vite-plugin-astro-server/error.ts:28]: If it doesn't have a name it should follow through to telemetry, shouldn't it? We would only need to be defensive if the error was null or undefined, which seems unlikely.

Comment (Princesseuh) [reply packages/astro/src/vite-plugin-astro-server/error.ts:28]: Oh yeah, for sure. I was just wondering if it can be `undefined` at all there or not

**Comment by bluwy** [comment]
I think Fred has the data if we filter the errors by `UnhandledRejection`, and we can check the kinds of errors we're recording before.

**Pull request merged**